### PR TITLE
Docs: Improve docs

### PR DESCRIPTION
**Pull request by kangmingtay**
## What kind of change does this PR introduce?
* Updates the docs for auth
* Add docs for `maybeSingle()` method to postgrest-js modifiers

## Auth docs updates
* signUp response for existing users
* `getUser()` method for retrieving an authenticated user from gotrue

### CODE REVIEW

**Review by phamhieu — APPROVED** [review]

File: web/spec/storage.json [hunk 1]
```diff
@@ -4440,7 +4440,7 @@
 												"isExported": true
 											},
 											"comment": {
-												"text": "An array of files to be deleted, including the path and file name. For example [`folder/image.png`].\n"
+												"text": "An array of files to be deletes, including the path and file name. For example [`folder/image.png`].\n"
```
Comment (phamhieu) [inline web/spec/storage.json:4443]: `An array of files to be deletes` looks strange to me

**Review by phamhieu — CHANGES_REQUESTED** [review]
need grammar check

File: web/spec/storage.json (hunk 1 above)
Comment (kangmingtay) [reply web/spec/storage.json:4443]: I think this is pulled from the storage docs, it got updated when i ran the make command

**Review by awalias — APPROVED** [review]

File: web/spec/supabase.yml [hunk 2]
```diff
@@ -565,6 +571,8 @@ pages:
   auth.user():
     title: 'user()'
     $ref: '@supabase/gotrue-js.GoTrueClient.user'
+    notes: |
+      This method gets the user object stored in local storage.
```
Comment (kiwicopple) [inline web/spec/supabase.yml:575]: @kangmingtay - this part is technically not true, as it gets it from memory (eg, `this.user`). LocalStorage is actually optional when initiating the client (or a dev can provider their own storage)

**Review by kiwicopple — APPROVED** [review]
@kangmingtay - I left one comment. Otherwise this looks great 👍

File: web/spec/supabase.yml (hunk 2 above)
Comment (kangmingtay) [reply web/spec/supabase.yml:575]: you're right! (https://github.com/supabase/gotrue-js/blob/master/src/GoTrueClient.ts#L296-L298) haha will make the update

**Review by kiwicopple — APPROVED** [review]

File: web/spec/storage.json (hunk 1 above)
Comment (kiwicopple) [reply web/spec/storage.json]: yeah, we should update the code comments inside storage-js to solve this

**Pull request merged**